### PR TITLE
redesign Clear benchmarks into CreateAddAndClear benchmarks

### PR DESCRIPTION
**Pull request by adamsitnik**
it's hard to benchmark a method that requires a clean state, takes nanoseconds to run and much more time to setup. Good example is `List<ValueType>.Clear` which just sets the list index to 0 and is forced to inline:

https://github.com/dotnet/coreclr/blob/8fa6bd4aaa949bbfd3dd263e7327996ab8fa7966/src/System.Private.CoreLib/shared/System/Collections/Generic/List.cs#L295-L311

I think that the best thing we can do with this problem right now is to redesign `Clear` benchmarks into `CreateAddAndClear`. If `Clear` is going to regress, the benchmark is still going to show it. 

@billwert if you like this idea I am going to redesign the `Remove` benchmarks in the same way

### CODE REVIEW

**Comment by adamsitnik** [comment]
Sample results:

```cmd
dotnet run -c Release -f netcoreapp2.2 --filter *CreateAddAndClear* --join --runtimes netcoreapp2.2 netcoreapp3.0
```

``` ini

BenchmarkDotNet=v0.11.3.1003-nightly, OS=Windows 10.0.18362
Intel Xeon CPU E5-1650 v4 3.60GHz, 1 CPU, 12 logical and 6 physical cores
.NET Core SDK=3.0.100-preview7-012697
  [Host]     : .NET Core 2.2.0 (CoreCLR 4.6.27110.04, CoreFX 4.6.27110.04), 64bit RyuJIT
  Job-YVBKFX : .NET Core 2.2.0 (CoreCLR 4.6.27110.04, CoreFX 4.6.27110.04), 64bit RyuJIT
  Job-PHMLPE : .NET Core 3.0.0-preview7-27826-20 (CoreCLR 4.700.19.32603, CoreFX 4.700.19.32613), 64bit RyuJIT

Runtime=Core  IterationTime=250.0000 ms  MaxIterationCount=20  
MinIterationCount=15  WarmupCount=1  

```
|                      Type |               Method |     Toolchain | Size |         Mean |        Error |       StdDev |       Median |          Min |          Max | Ratio | RatioSD | Gen 0/1k Op | Gen 1/1k Op | Gen 2/1k Op | Allocated Memory/Op |
|-------------------------- |--------------------- |-------------- |----- |-------------:|-------------:|-------------:|-------------:|-------------:|-------------:|------:|--------:|------------:|------------:|------------:|--------------------:|
|  CreateAddAndClear&lt;Int32&gt; |                Array | netcoreapp2.2 |  512 |     496.6 ns |     5.424 ns |     5.074 ns |     496.1 ns |     489.6 ns |     509.1 ns |  1.00 |    0.00 |      0.3286 |      0.0020 |           - |             2.02 KB |
|  CreateAddAndClear&lt;Int32&gt; |                Array | netcoreapp3.0 |  512 |     493.3 ns |     6.002 ns |     5.321 ns |     493.9 ns |     483.0 ns |     502.9 ns |  0.99 |    0.01 |      0.2639 |      0.0020 |           - |             2.02 KB |
|                           |                      |               |      |              |              |              |              |              |              |       |         |             |             |             |                     |
| CreateAddAndClear&lt;String&gt; |                Array | netcoreapp2.2 |  512 |   1,818.3 ns |    12.973 ns |    12.135 ns |   1,820.0 ns |   1,796.8 ns |   1,834.6 ns |  1.00 |    0.00 |      0.6522 |      0.0072 |           - |             4.02 KB |
| CreateAddAndClear&lt;String&gt; |                Array | netcoreapp3.0 |  512 |   1,834.7 ns |    16.021 ns |    14.202 ns |   1,836.2 ns |   1,811.1 ns |   1,865.7 ns |  1.01 |    0.01 |      0.5191 |      0.0073 |           - |             4.02 KB |
|                           |                      |               |      |              |              |              |              |              |              |       |         |             |             |             |                     |
|  CreateAddAndClear&lt;Int32&gt; |                 Span | netcoreapp2.2 |  512 |     600.3 ns |     4.684 ns |     4.152 ns |     599.5 ns |     595.6 ns |     608.6 ns |  1.00 |    0.00 |      0.3277 |           - |           - |             2.02 KB |
|  CreateAddAndClear&lt;Int32&gt; |                 Span | netcoreapp3.0 |  512 |     636.9 ns |    36.152 ns |    41.633 ns |     647.2 ns |     589.2 ns |     703.5 ns |  1.09 |    0.06 |      0.2620 |           - |           - |             2.02 KB |
|                           |                      |               |      |              |              |              |              |              |              |       |         |             |             |             |                     |
| CreateAddAndClear&lt;String&gt; |                 Span | netcoreapp2.2 |  512 |   1,671.1 ns |    14.506 ns |    13.569 ns |   1,669.0 ns |   1,649.6 ns |   1,699.1 ns |  1.00 |    0.00 |      0.6464 |      0.0067 |           - |             4.02 KB |
| CreateAddAndClear&lt;String&gt; |                 Span | netcoreapp3.0 |  512 |   1,806.2 ns |    17.725 ns |    16.580 ns |   1,804.6 ns |   1,782.9 ns |   1,845.0 ns |  1.08 |    0.01 |      0.5232 |      0.0074 |           - |             4.02 KB |
|                           |                      |               |      |              |              |              |              |              |              |       |         |             |             |             |                     |
|  CreateAddAndClear&lt;Int32&gt; |                 List | netcoreapp2.2 |  512 |   1,369.5 ns |    11.511 ns |    10.767 ns |   1,368.5 ns |   1,355.8 ns |   1,390.5 ns |  1.00 |    0.00 |      0.6815 |      0.0055 |           - |             4.21 KB |
|  CreateAddAndClear&lt;Int32&gt; |                 List | netcoreapp3.0 |  512 |   1,375.0 ns |    17.731 ns |    16.586 ns |   1,374.4 ns |   1,348.9 ns |   1,407.5 ns |  1.00 |    0.02 |      0.5475 |      0.0054 |           - |              4.2 KB |
|                           |                      |               |      |              |              |              |              |              |              |       |         |             |             |             |                     |
| CreateAddAndClear&lt;String&gt; |                 List | netcoreapp2.2 |  512 |   2,650.9 ns |    21.488 ns |    19.048 ns |   2,653.2 ns |   2,624.9 ns |   2,690.0 ns |  1.00 |    0.00 |      1.3241 |      0.0320 |           - |              8.2 KB |
| CreateAddAndClear&lt;String&gt; |                 List | netcoreapp3.0 |  512 |   2,744.3 ns |    16.448 ns |    14.581 ns |   2,746.4 ns |   2,719.4 ns |   2,773.5 ns |  1.04 |    0.01 |      1.0616 |      0.0332 |           - |             8.19 KB |
|                           |                      |               |      |              |              |              |              |              |              |       |         |             |             |             |                     |
|  CreateAddAndClear&lt;Int32&gt; |           LinkedList | netcoreapp2.2 |  512 |  11,211.5 ns |    72.765 ns |    68.064 ns |  11,223.9 ns |  11,089.0 ns |  11,298.1 ns |  1.00 |    0.00 |      3.9063 |      0.1347 |           - |            24.08 KB |
|  CreateAddAndClear&lt;Int32&gt; |           LinkedList | netcoreapp3.0 |  512 |  10,936.1 ns |   106.307 ns |    99.439 ns |  10,957.0 ns |  10,793.0 ns |  11,125.4 ns |  0.98 |    0.01 |      3.1381 |      0.1308 |           - |            24.07 KB |
|                           |                      |               |      |              |              |              |              |              |              |       |         |             |             |             |                     |
| CreateAddAndClear&lt;String&gt; |           LinkedList | netcoreapp2.2 |  512 |  19,793.1 ns |   164.805 ns |   154.159 ns |  19,714.6 ns |  19,616.1 ns |  20,151.4 ns |  1.00 |    0.00 |      3.8717 |      0.0790 |           - |            24.08 KB |
| CreateAddAndClear&lt;String&gt; |           LinkedList | netcoreapp3.0 |  512 |  11,730.5 ns |    81.217 ns |    71.997 ns |  11,726.0 ns |  11,592.8 ns |  11,859.4 ns |  0.59 |    0.01 |      3.1227 |      0.0932 |           - |            24.07 KB |
|                           |                      |               |      |              |              |              |              |              |              |       |         |             |             |             |                     |
|  CreateAddAndClear&lt;Int32&gt; |              HashSet | netcoreapp2.2 |  512 |  12,762.5 ns |   163.120 ns |   152.582 ns |  12,710.9 ns |  12,541.9 ns |  13,024.7 ns |  1.00 |    0.00 |      4.3628 |      0.4058 |           - |            27.06 KB |
|  CreateAddAndClear&lt;Int32&gt; |              HashSet | netcoreapp3.0 |  512 |  13,147.4 ns |   128.759 ns |   120.441 ns |  13,172.8 ns |  12,984.4 ns |  13,351.6 ns |  1.03 |    0.01 |      3.4898 |      0.3701 |           - |            27.06 KB |
|                           |                      |               |      |              |              |              |              |              |              |       |         |             |             |             |                     |
| CreateAddAndClear&lt;String&gt; |              HashSet | netcoreapp2.2 |  512 |  29,804.2 ns |   221.186 ns |   196.075 ns |  29,785.2 ns |  29,467.9 ns |  30,264.6 ns |  1.00 |    0.00 |      5.3776 |      0.5975 |           - |            33.67 KB |
| CreateAddAndClear&lt;String&gt; |              HashSet | netcoreapp3.0 |  512 |  26,632.7 ns |   314.307 ns |   294.003 ns |  26,536.4 ns |  26,256.5 ns |  27,183.6 ns |  0.89 |    0.01 |      4.3878 |      0.5351 |           - |            33.67 KB |
|                           |                      |               |      |              |              |              |              |              |              |       |         |             |             |             |                     |
|  CreateAddAndClear&lt;Int32&gt; |           Dictionary | netcoreapp2.2 |  512 |  14,616.0 ns |    75.905 ns |    71.002 ns |  14,631.7 ns |  14,488.3 ns |  14,711.3 ns |  1.00 |    0.00 |      5.4499 |      0.6378 |           - |            33.69 KB |
|  CreateAddAndClear&lt;Int32&gt; |           Dictionary | netcoreapp3.0 |  512 |  13,680.5 ns |   146.534 ns |   137.068 ns |  13,691.3 ns |  13,401.9 ns |  13,876.9 ns |  0.94 |    0.01 |      4.3801 |      0.5054 |           - |            33.68 KB |
|                           |                      |               |      |              |              |              |              |              |              |       |         |             |             |             |                     |
| CreateAddAndClear&lt;String&gt; |           Dictionary | netcoreapp2.2 |  512 |  30,552.9 ns |   240.771 ns |   225.218 ns |  30,566.1 ns |  29,991.4 ns |  30,905.7 ns |  1.00 |    0.00 |      7.5243 |      1.2136 |           - |            46.97 KB |
| CreateAddAndClear&lt;String&gt; |           Dictionary | netcoreapp3.0 |  512 |  26,186.9 ns |   339.337 ns |   317.416 ns |  26,201.2 ns |  25,719.6 ns |  26,644.3 ns |  0.86 |    0.01 |      6.1153 |      0.9328 |           - |            46.96 KB |
|                           |                      |               |      |              |              |              |              |              |              |       |         |             |             |             |                     |
|  CreateAddAndClear&lt;Int32&gt; |           SortedList | netcoreapp2.2 |  512 |  61,455.5 ns |   386.470 ns |   361.504 ns |  61,427.7 ns |  60,634.4 ns |  62,054.0 ns |  1.00 |    0.00 |      1.2160 |           - |           - |             8.41 KB |
|  CreateAddAndClear&lt;Int32&gt; |           SortedList | netcoreapp3.0 |  512 |  58,566.0 ns |   307.862 ns |   287.974 ns |  58,571.3 ns |  58,020.3 ns |  58,966.6 ns |  0.95 |    0.01 |      0.9470 |           - |           - |             8.41 KB |
|                           |                      |               |      |              |              |              |              |              |              |       |         |             |             |             |                     |
| CreateAddAndClear&lt;String&gt; |           SortedList | netcoreapp2.2 |  512 | 594,960.1 ns | 3,742.404 ns | 3,317.543 ns | 595,518.4 ns | 589,614.2 ns | 601,493.3 ns |  1.00 |    0.00 |           - |           - |           - |            16.38 KB |
| CreateAddAndClear&lt;String&gt; |           SortedList | netcoreapp3.0 |  512 | 484,609.4 ns | 4,332.811 ns | 4,052.914 ns | 485,838.2 ns | 478,274.4 ns | 493,297.3 ns |  0.81 |    0.01 |      1.8939 |           - |           - |            16.38 KB |
|                           |                      |               |      |              |              |              |              |              |              |       |         |             |             |             |                     |
|  CreateAddAndClear&lt;Int32&gt; |            SortedSet | netcoreapp2.2 |  512 |  46,548.4 ns |   524.449 ns |   490.570 ns |  46,578.9 ns |  45,912.3 ns |  47,722.5 ns |  1.00 |    0.00 |      3.2100 |      0.1888 |           - |            20.05 KB |
|  CreateAddAndClear&lt;Int32&gt; |            SortedSet | netcoreapp3.0 |  512 |  46,057.4 ns |   448.369 ns |   419.404 ns |  46,016.9 ns |  45,398.3 ns |  46,905.2 ns |  0.99 |    0.01 |      2.5964 |      0.1855 |           - |            20.05 KB |
|                           |                      |               |      |              |              |              |              |              |              |       |         |             |             |             |                     |
| CreateAddAndClear&lt;String&gt; |            SortedSet | netcoreapp2.2 |  512 | 582,966.2 ns | 1,354.774 ns | 1,200.972 ns | 582,814.0 ns | 580,324.0 ns | 584,912.0 ns |  1.00 |    0.00 |      2.3148 |           - |           - |            24.05 KB |
| CreateAddAndClear&lt;String&gt; |            SortedSet | netcoreapp3.0 |  512 | 487,076.9 ns | 2,736.607 ns | 2,425.931 ns | 487,976.2 ns | 481,664.1 ns | 489,311.8 ns |  0.84 |    0.01 |      1.8939 |           - |           - |            24.05 KB |
|                           |                      |               |      |              |              |              |              |              |              |       |         |             |             |             |                     |
|  CreateAddAndClear&lt;Int32&gt; |     SortedDictionary | netcoreapp2.2 |  512 |  74,394.0 ns |   417.071 ns |   348.273 ns |  74,379.5 ns |  73,853.0 ns |  75,127.6 ns |  1.00 |    0.00 |      3.8325 |           - |           - |            24.12 KB |
|  CreateAddAndClear&lt;Int32&gt; |     SortedDictionary | netcoreapp3.0 |  512 |  73,258.5 ns |   845.295 ns |   790.690 ns |  73,118.2 ns |  72,394.4 ns |  74,879.7 ns |  0.98 |    0.01 |      2.9621 |           - |           - |            24.11 KB |
|                           |                      |               |      |              |              |              |              |              |              |       |         |             |             |             |                     |
| CreateAddAndClear&lt;String&gt; |     SortedDictionary | netcoreapp2.2 |  512 | 648,767.8 ns | 4,039.100 ns | 3,580.557 ns | 649,456.3 ns | 641,538.7 ns | 653,278.1 ns |  1.00 |    0.00 |      2.6042 |           - |           - |            28.12 KB |
| CreateAddAndClear&lt;String&gt; |     SortedDictionary | netcoreapp3.0 |  512 | 522,969.5 ns | 2,914.114 ns | 2,725.864 ns | 523,625.7 ns | 516,622.1 ns | 526,561.9 ns |  0.81 |    0.01 |      2.0833 |           - |           - |            28.11 KB |
|                           |                      |               |      |              |              |              |              |              |              |       |         |             |             |             |                     |
|  CreateAddAndClear&lt;Int32&gt; | ConcurrentDictionary | netcoreapp2.2 |  512 |  85,312.3 ns |   994.541 ns |   930.295 ns |  85,387.6 ns |  83,918.9 ns |  87,348.1 ns |  1.00 |    0.00 |     20.0408 |      4.0761 |           - |           124.39 KB |
|  CreateAddAndClear&lt;Int32&gt; | ConcurrentDictionary | netcoreapp3.0 |  512 |  82,668.5 ns |   704.464 ns |   624.489 ns |  82,700.5 ns |  81,533.7 ns |  83,820.3 ns |  0.97 |    0.01 |     15.9574 |      2.9920 |           - |           124.39 KB |
|                           |                      |               |      |              |              |              |              |              |              |       |         |             |             |             |                     |
| CreateAddAndClear&lt;String&gt; | ConcurrentDictionary | netcoreapp2.2 |  512 | 136,656.6 ns | 1,111.743 ns |   985.531 ns | 136,442.7 ns | 135,478.8 ns | 138,791.3 ns |  1.00 |    0.00 |     20.6522 |      3.8043 |           - |           129.06 KB |
| CreateAddAndClear&lt;String&gt; | ConcurrentDictionary | netcoreapp3.0 |  512 | 103,268.8 ns |   771.323 ns |   721.496 ns | 103,134.0 ns | 102,252.4 ns | 104,539.4 ns |  0.76 |    0.01 |     17.2697 |      3.7007 |           - |           132.58 KB |
|                           |                      |               |      |              |              |              |              |              |              |       |         |             |             |             |                     |
|  CreateAddAndClear&lt;Int32&gt; |                Stack | netcoreapp2.2 |  512 |   1,617.8 ns |     6.354 ns |     4.961 ns |   1,616.3 ns |   1,612.0 ns |   1,624.7 ns |  1.00 |    0.00 |      0.6828 |      0.0065 |           - |             4.21 KB |
|  CreateAddAndClear&lt;Int32&gt; |                Stack | netcoreapp3.0 |  512 |   1,543.9 ns |    22.180 ns |    19.662 ns |   1,540.4 ns |   1,522.6 ns |   1,586.0 ns |  0.96 |    0.01 |      0.5487 |      0.0062 |           - |              4.2 KB |
|                           |                      |               |      |              |              |              |              |              |              |       |         |             |             |             |                     |
| CreateAddAndClear&lt;String&gt; |                Stack | netcoreapp2.2 |  512 |   3,483.6 ns |    26.457 ns |    24.748 ns |   3,484.0 ns |   3,443.2 ns |   3,543.6 ns |  1.00 |    0.00 |      1.3271 |      0.0279 |           - |              8.2 KB |
| CreateAddAndClear&lt;String&gt; |                Stack | netcoreapp3.0 |  512 |   3,321.0 ns |    35.413 ns |    31.393 ns |   3,326.3 ns |   3,269.8 ns |   3,363.5 ns |  0.95 |    0.01 |      1.0613 |      0.0265 |           - |             8.19 KB |
|                           |                      |               |      |              |              |              |              |              |              |       |         |             |             |             |                     |
|  CreateAddAndClear&lt;Int32&gt; |      ConcurrentStack | netcoreapp2.2 |  512 |   7,460.5 ns |    63.634 ns |    59.523 ns |   7,453.2 ns |   7,350.6 ns |   7,571.9 ns |  1.00 |    0.00 |      2.5734 |      0.0592 |           - |            16.02 KB |
|  CreateAddAndClear&lt;Int32&gt; |      ConcurrentStack | netcoreapp3.0 |  512 |   6,643.4 ns |    90.001 ns |    84.187 ns |   6,616.8 ns |   6,498.8 ns |   6,798.3 ns |  0.89 |    0.01 |      2.0825 |      0.0527 |           - |            16.02 KB |
|                           |                      |               |      |              |              |              |              |              |              |       |         |             |             |             |                     |
| CreateAddAndClear&lt;String&gt; |      ConcurrentStack | netcoreapp2.2 |  512 |  16,687.2 ns |   172.478 ns |   161.336 ns |  16,741.5 ns |  16,459.8 ns |  16,950.2 ns |  1.00 |    0.00 |      2.5412 |      0.0687 |           - |            16.02 KB |
| CreateAddAndClear&lt;String&gt; |      ConcurrentStack | netcoreapp3.0 |  512 |   7,173.7 ns |   112.866 ns |   105.575 ns |   7,156.1 ns |   7,016.8 ns |   7,409.6 ns |  0.43 |    0.01 |      2.0871 |      0.0572 |           - |            16.02 KB |
|                           |                      |               |      |              |              |              |              |              |              |       |         |             |             |             |                     |
|  CreateAddAndClear&lt;Int32&gt; |                Queue | netcoreapp2.2 |  512 |   2,256.2 ns |    26.238 ns |    23.259 ns |   2,256.7 ns |   2,229.3 ns |   2,314.7 ns |  1.00 |    0.00 |      0.6792 |      0.0091 |           - |             4.22 KB |
|  CreateAddAndClear&lt;Int32&gt; |                Queue | netcoreapp3.0 |  512 |   2,257.4 ns |    28.012 ns |    26.202 ns |   2,251.8 ns |   2,216.9 ns |   2,308.6 ns |  1.00 |    0.01 |      0.5476 |      0.0090 |           - |             4.21 KB |
|                           |                      |               |      |              |              |              |              |              |              |       |         |             |             |             |                     |
| CreateAddAndClear&lt;String&gt; |                Queue | netcoreapp2.2 |  512 |   4,252.9 ns |    18.659 ns |    15.581 ns |   4,258.0 ns |   4,210.9 ns |   4,266.3 ns |  1.00 |    0.00 |      1.3262 |      0.0340 |           - |              8.2 KB |
| CreateAddAndClear&lt;String&gt; |                Queue | netcoreapp3.0 |  512 |   3,573.3 ns |    42.088 ns |    35.145 ns |   3,578.5 ns |   3,503.0 ns |   3,631.5 ns |  0.84 |    0.01 |      1.0641 |      0.0284 |           - |              8.2 KB |
|                           |                      |               |      |              |              |              |              |              |              |       |         |             |             |             |                     |
|  CreateAddAndClear&lt;Int32&gt; |      ConcurrentQueue | netcoreapp2.2 |  512 |   7,902.8 ns |    56.164 ns |    52.536 ns |   7,900.9 ns |   7,796.7 ns |   7,999.9 ns |  1.00 |    0.00 |      1.7003 |      0.0630 |           - |            10.69 KB |
|  CreateAddAndClear&lt;Int32&gt; |      ConcurrentQueue | netcoreapp3.0 |  512 |   7,938.8 ns |   101.654 ns |    95.087 ns |   7,930.1 ns |   7,780.8 ns |   8,091.4 ns |  1.00 |    0.01 |      1.2392 |      0.0318 |           - |             9.56 KB |
|                           |                      |               |      |              |              |              |              |              |              |       |         |             |             |             |                     |
| CreateAddAndClear&lt;String&gt; |      ConcurrentQueue | netcoreapp2.2 |  512 |   9,069.6 ns |   129.994 ns |   121.597 ns |   9,048.4 ns |   8,918.2 ns |   9,304.5 ns |  1.00 |    0.00 |      3.0003 |      0.1807 |           - |            18.69 KB |
| CreateAddAndClear&lt;String&gt; |      ConcurrentQueue | netcoreapp3.0 |  512 |   8,614.5 ns |    88.689 ns |    82.960 ns |   8,628.7 ns |   8,478.5 ns |   8,765.5 ns |  0.95 |    0.02 |      2.2908 |      0.1368 |           - |            17.56 KB |

File: src/benchmarks/micro/corefx/System.Collections/Remove/CreateAddAndClear.cs [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+            for (int i = 0; i < _keys.Length; i++)
+            {
+                span[i] = _keys[i];
+            }
+            span.Clear();
+            return span;
+        }
+
+        [Benchmark]
+        public List<T> List()
```
Comment (billwert) [inline src/benchmarks/micro/corefx/System.Collections/Remove/CreateAddAndClear.cs]: Missing `Clear`?

File: src/benchmarks/micro/corefx/System.Collections/Remove/CreateAddAndClear.cs [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+            foreach (T uniqueKey in _keys)
+            {
+                sortedList.Add(uniqueKey, uniqueKey);
+            }
+            sortedList.Clear();
+            return sortedList;
+        }
+
+        [Benchmark]
+        public SortedSet<T> SortedSet()
```
Comment (billwert) [inline src/benchmarks/micro/corefx/System.Collections/Remove/CreateAddAndClear.cs]: missing `Clear`?

File: src/benchmarks/micro/corefx/System.Collections/Remove/CreateAddAndClear.cs (hunk 1 above)
Comment (adamsitnik) [reply src/benchmarks/micro/corefx/System.Collections/Remove/CreateAddAndClear.cs]: good catch! I have no idea how I could have missed it.

**Comment by adamsitnik** [comment]
> Do we have separate "Create" benchmarks for all of these? Can we delete those also?

Good point. We have:
https://github.com/dotnet/performance/blob/7a461549f50882e781a1b98eb778c20491e78a3b/src/benchmarks/micro/corefx/System.Collections/Add/AddDefaultSize.cs#L17

I have now removed these duplicated benchmarks

**Comment by adamsitnik** [comment]
I've fixed the issues pointed by @billwert but since Bill is going to be offline for the next few days and I really want to get rid of this unstable benchmark to continue my perf investigation I am going to merge this PR now.

**Pull request merged**